File: 014. Longest Common Prefix/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) return string();
        int i, j;
        for (i = 0; i < strs[0].length(); i++) 
            for (j = 1; j < strs.size(); j++) 
                if (strs[j][i] != strs[0][i])
                    return strs[0].substr(0, i);
        return strs[0].substr(0, i);
    }
};

File: 014. Longest Common Prefix/14. Longest Common Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        prefix = ""
        try:  # if not beyond the boundary
            i = 0
            while 1: 
                if all(s[i] == strs[0][i] for s in strs):  # use first element as a test
                    prefix += strs[0][i]
                    i += 1
                else:
                    return prefix
        except IndexError:
            return prefix
